includes/Core/User_Input/User_Input.php

```diff
@@ -0,0 +1,384 @@
+				continue;
+			}
+
+			$setting['author'] = array(
+				'photo' => get_avatar_url( $answered_by ),
+				'name'  => get_the_author_meta( 'user_email', $answered_by ),
+			);
+		}
+
+		// If there are un-answered questions, return default empty values for them.
+		foreach ( $questions as $question_key => $question_value ) {
+			if ( ! isset( $settings[ $question_key ] ) ) {
+				$settings[ $question_key ]           = $question_value;
+				$settings[ $question_key ]['values'] = array();
+			}
+		}
+
+		return $settings;
+	}
+
+	/**
+	 * Determines whether the current user input settings have empty values or not.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $settings The settings to check.
+	 * @return boolean|null TRUE if at least one of the settings has empty values, otherwise FALSE.
+	 */
+	public function are_settings_empty( $settings = array() ) {
+		if ( empty( $settings ) ) {
+			$settings = $this->get_answers();
+
+			if ( is_wp_error( $settings ) ) {
+				return null;
+			}
+		}
+
+		$empty_settings = array_filter(
+			$settings,
+			function( $setting ) {
+				return empty( $setting['values'] );
+			}
+		);
+
+		return 0 < count( $empty_settings );
+	}
+
+	/**
+	 * Sets user input settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $settings User settings.
+	 * @return array|WP_Error User input settings.
+	 */
+	public function set_answers( $settings ) {
+		$questions     = static::$questions;
+		$site_settings = array();
+		$user_settings = array();
+
+		foreach ( $settings as $setting_key => $answers ) {
+			$setting_data           = array();
+			$setting_data['values'] = $answers;
+			$setting_data['scope']  = $questions[ $setting_key ]['scope'];
+
+			if ( 'site' === $setting_data['scope'] ) {
+				$setting_data['answeredBy']    = get_current_user_id();
+				$site_settings[ $setting_key ] = $setting_data;
+			} elseif ( 'user' === $setting_data['scope'] ) {
+				$user_settings[ $setting_key ] = $setting_data;
+			}
+		}
+
+		$this->site_specific_questions->set( $site_settings );
+		$this->user_specific_questions->set( $user_settings );
+
+		$updated_settings = $this->get_answers();
+		$is_empty         = $this->are_settings_empty( $updated_settings );
+
+		if ( ! is_null( $is_empty ) ) {
+			$this->authentication->get_user_input_state()->set( $is_empty ? User_Input_State::VALUE_MISSING : User_Input_State::VALUE_COMPLETED );
+		}
+
+		return $updated_settings;
+	}
+
+	/**
+	 * Method re-usable by scoped classes to sanitize settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Site_Specific_Questions|User_Specific_Questions $setting  Instance of either setting class.
+	 * @param array                                           $settings Settings to sanitize.
+	 * @param string                                          $scope    Scope of the settings, either 'site' or 'user'.
+	 * @return array Sanitized settings.
+	 */
+	public static function sanitize_settings( $setting, $settings, $scope ) {
```

Comment:
See my above suggestion above. Do you think it's possible to make this code simpler to follow somehow? I don't have any specific suggestion, but it feels hard to follow to me. :slightly_smiling_face: